django/db/models/deletion.py

```diff
@@ -51,6 +52,23 @@ def DO_NOTHING(collector, field, sub_objs, using):
     pass
 
 
+def get_candidate_relations_to_delete(opts):
+    # Collect models that contain candidate relations to delete. This may include
+    # relations coming from proxy models.
+    candidate_models = {opts}
+    candidate_models = candidate_models.union(opts.concrete_model._meta.proxied_children)
+    # For each model, get all candidate fields.
+    candidate_model_fields = chain.from_iterable(
+        opts.get_fields(include_hidden=True) for opts in candidate_models
+    )
+    # Keep related objects not coming from ManyToManyFields from all candidate
+    # model fields.
+    return (
+        f for f in candidate_model_fields
+        if f.auto_created and not f.concrete and not f.many_to_many
```

Comment:
@freakboy3742 I agree that "for now" `GenericRelation` is just a lookup aid, but ideally it shouldn't be just that. Eventually we want all reverse relation to be actual fields (note that in case of `GenericRelation` we may need to introduce a new field to preserve backwards compatibility for people who really just wanted "lookup aid"). If I made a M2M backed M2O to avoid tight coupling 2 apps (i.e. you can't control a third party app model), I'd like it to be included in deletion. It would also make sense in the context of a "reverse O2O" such as https://github.com/akaariai/django-reverse-unique/tree/master/reverse_unique.